src/WidgetBase.php

```diff
@@ -191,7 +211,51 @@ public function setWeight($weight) {
   /**
    * {@inheritdoc}
    */
-  public function validate(array &$form, FormStateInterface $form_state) {}
+  public function prepareEntities(FormStateInterface $form_state) {
+    return [];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validate(array &$form, FormStateInterface $form_state) {
+    $trigger = $form_state->getTriggeringElement();
+
+    if (in_array('submit', $trigger['#array_parents'])) {
+      $entities = $this->prepareEntities($form_state);
+      $violations = $this->runWidgetValidators($entities);
+      if (!empty($violations)) {
+        /** @var \Symfony\Component\Validator\ConstraintViolationListInterface $violation */
+        foreach ($violations as $violation) {
+          $form_state->setError($form['widget']['upload'], $violation->getMessage());
```

Comment:
This element is Upload widget specific. Since this is base class it should be general.